src/module.c

```diff
@@ -2890,8 +2890,11 @@ void RM_ReplySetSetLength(RedisModuleCtx *ctx, long len) {
 }
 
 /* Very similar to RedisModule_ReplySetMapLength
- * Visit https://github.com/antirez/RESP3/blob/master/spec.md for more info about RESP3. */
+ * Visit https://github.com/antirez/RESP3/blob/master/spec.md for more info about RESP3.
+ *
+ * Not supported by RESP2 and will return void. */
 void RM_ReplySetAttributeLength(RedisModuleCtx *ctx, long len) {
+    if (ctx->client->resp == 2) return;
```

Comment:
correct me if i'm wrong, but this check is only needed if the module ignored the error that returned from the previous call to RM_ReplyWithAttribute, right?

i'm trying to decide if this fix should be back-ported, or mentioned in the docs as something not all versions had, or can we just regard it as a minor issue, considering it only affects modules that ignored a previous error.

@MeirShpilraien WDYT?

Reply:
@oranagra yes it will only happened with a misbehave module so I do not think its urgent to back-port it. I think that we can document it as undefined behaviour and that it should not be done, this way there is no guarantee what will happened on each Redis version. WDYT?

Reply:
updated the comment to be a bit more vague, i.e describe what should be done, and not what is the expected outcome.
@DarrenJiang13 @MeirShpilraien let me know it it looks ok, and if you think we can consider this a minor improvement and not mention it anywhere..

Reply:
LGTM and I yes I think this can be considered as a minor improvement.

Reply:
LGTM. Yes I think the new comment is clear enough and no need to mention in another place.